File: single-language-analysis/dataset-files/java/TestSmell052.java
// Original URL: https://github.com/elaissoussi/exercices/blob/50ac9b6d9e89c094010a9ed637bbd3198f36a729/marsrover/MarsRoverTest.java#L81-L96

public class TestSmell052 {
	@Test
	@Ignore
	public void pathOverlapsAreMarkedWithRightCursor() {
		MarsRover marsRover = new MarsRover("ssssssrsss").turnRight()
				.moveForward().moveForward().moveForward().turnRight()
				.moveForward().moveForward().moveForward().moveForward();

		String expectedPath = new StringBuilder().append("   *   ")
				.append(LINE_SEPARATOR).append("X--+--+")
				.append(LINE_SEPARATOR).append("   |  |")
				.append(LINE_SEPARATOR).append("   |  |")
				.append(LINE_SEPARATOR).append("   +--+")
				.append(LINE_SEPARATOR).toString();

		assertThat(marsRover.path(), equalTo(expectedPath));
	}
}

File: single-language-analysis/dataset-files/python/test_smell_052.py
import pytest

class TestMarsRover:
    @pytest.mark.skip
    def test_smell_052(self):
        mars_rover = MarsRover("ssssssrsss").turn_right() \
            .move_forward().move_forward().move_forward().turn_right() \
            .move_forward().move_forward().move_forward().move_forward()

        expected_path = "   *   \nX--+--+\n   |  |\n   |  |\n   +--+\n"

        assert mars_rover.path() == expected_path